File: support/tests/bad/badtracker.py
#!/usr/bin/python

from trax import *
import time

region = None
break_time = 0
with Server([Region.RECTANGLE], [Image.MEMORY, Image.PATH], verbose=True) as server:
    while True:
        request = server.wait()
        if request.type in [TraxStatus.QUIT, TraxStatus.ERROR]:
            break
        if request.type == TraxStatus.INITIALIZE:
            region = request.region
            break_time = float(request.properties.get("break_time", "5"))
        server.status(region)
		break_time--;
        if break_time <= 0:
            raise Exception("Bad tracker")

File: support/tests/bad/badtracker.m
function [] = badtracker()

traxserver('setup', 'polygon', {'path', 'memory', 'buffer'});

memory = [0 0 0 0];

when = 10;
cause = 'abort';

while 1

    [image, region, props] = traxserver('wait', 'PropStruct', true);

    if isempty(image)
		break;
	end;

	if ~isempty(region)
        memory = region;

		if ~isempty(props)
			if isfield(props, 'break_type')
				cause = props.break_type;
			end;
			if isfield(props, 'break_time')
				when = str2double(props.break_time);
			end;
		end;

    end

	if when <= 0
		badmex(cause);
		break;
	end;

	when = when - 1;

	traxserver('status', memory);
end

traxserver('quit');
